### ISSUE TITLE
 clone with non-empty argument not supported

### ISSUE THREAD

**Issue by marcoeg** [harshgondaliya/BurstRadar#2]
In compiling burstradar I am getting the following warning:

`burstradar.p4(192): [--Wwarn=unsupported] warning: clone3: clone with non-empty argument not supported`

A Google search pointed me to:

[https://github.com/p4lang/p4c/pull/2000](https://github.com/p4lang/p4c/pull/2000)

As it appears that the bmv2 back-end currently does not implement correctly recirculate, resubmit and clone3.

**Comment by harshgondaliya** [comment]
Yes, bmv2 back-end currently does not implement correctly recirculate, resubmit and clone3.

The real implementation of BurstRadar is on the Barefoot Tofino switching ASIC as explained in the original [paper](https://drive.google.com/file/d/1hZJ0YdIJRiReifpX6AFvgaDvWzmEaDu0/view) by the authors. Tofino architecture implements the clone operation properly.

The work put up in this repository has adapted the original BurstRadar system (explained in the paper) to make it work on v1model architecture.
